src/components/Profile/LanguageSection.jsx

```diff
@@ -2,45 +2,73 @@ import React from 'react'
 
 import { isFlagshipApp } from 'cozy-device-helper'
 import { useI18n } from 'cozy-ui/transpiled/react/I18n'
+import { useQuery } from 'cozy-client'
+import { useDispatch } from 'react-redux'
 
 import Select from 'components/Select'
 import { useSetLang } from 'hooks/useSetLang'
+import { useMutation } from 'hooks/useMutation'
+import { buildSettingsInstanceQuery } from 'lib/queries'
+import { SET_LANG } from 'actions'
 
 const LANG_OPTIONS = ['en', 'fr', 'es']
 
-const LanguageSection = props => {
+const LanguageSection = () => {
   const { t } = useI18n()
-  const { fields, onChange } = props
+  const dispatch = useDispatch()
+  const { mutate, isLoading, isSuccess, error } = useMutation({
+    onSuccess: data => {
+      dispatch({ type: SET_LANG, lang: data.locale })
```

Comment:
next pr, remove the need to dispatch something? :D

Reply:
Yes for sure